cmd/nomos/status/client.go

```diff
@@ -354,6 +361,29 @@ func (c *ClusterClient) namespaceRepoClusterStatus(ctx context.Context, cs *Clus
 	}
 }
 
+// IsOssInstallation will check for the existence of ConfigManagement object, Operator deployment, and RootSync CRD
+// If RootSync CRD exist but ConfigManagement and Operator doesn't, it indicates an OSS installation
+func (c *ClusterClient) IsOssInstallation(ctx context.Context, cs *ClusterState) bool {
+	rootSyncCRD := &apiextensionsv1.CustomResourceDefinition{}
+	repoSyncCRD := &apiextensionsv1.CustomResourceDefinition{}
+	rootSyncCRDErr := c.Client.Get(ctx, client.ObjectKey{Name: rootSyncCRDName}, rootSyncCRD)
+	repoSyncCRDErr := c.Client.Get(ctx, client.ObjectKey{Name: repoSyncCRDName}, repoSyncCRD)
+	if apierrors.IsNotFound(rootSyncCRDErr) && apierrors.IsNotFound(repoSyncCRDErr) {
+		cs.Error = "RootSync CRD and RepoSync CRD are not found"
+		return false
+	}
```

Comment:
It is wrong to check the R*Sync CRD first as it might use the mono-repo mode.

You need to check the CM object and the operator Deployment first.

- If CM and operator exist => return false
- If CM or operator doesn't exist
   - If R*Sync CRD exist => return true to indicate OSS installation
   - else => return installation error
- If failed to get CM, or operator, or R*Sync CRD => return get error

Reply:
thanks Nan! The logic has been updated to follow your suggestion